melodies_monet/plots/surfplots.py

```diff
@@ -829,12 +830,168 @@ def calculate_boxplot(df, df_reg=None, column=None, label=None, plot_dict=None,
     else:
         comb_bx[label] = df[column]
     label_bx.append(plot_kwargs)
-    
+
     return comb_bx, label_bx
+
+####################################################################################
+#This is BEIMING modified 'calculate_multi_boxplot',added region_bx as another output
+#####################################################################################
+def calculate_multi_boxplot(df, df_reg=None, region_name= None,column=None, label=None, plot_dict=None, comb_bx = None, label_bx = None): 
+    """Combines data into acceptable format for box-plot
     
+    Parameters
+    ----------
+    df : pandas.DataFrame
+        model/obs paired data to plot
+    df_reg : pandas.DataFrame
+        model/obs paired regulatory data to plot
+    column : str
+        Column label of variable to plot
+    label : str
+        Name of variable to use in plot legend
+    comb_bx: dataframe
+        dataframe containing information to create box-plot from previous 
+        occurrence so can overlay multiple model results on plot
+    label_bx: list
+        list of string labels to use in box-plot from previous occurrence so 
+        can overlay multiple model results on plot
+    Returns
+    -------
+    dataframe, list, dataframe           
+        dataframe containing information to create box-plot
+        list of string labels to use in box-plot
+        dataframe containing informaiton for regions to help create multi-box-plot
+        
+    """
+    region_bx = pd.DataFrame()                   #BEIMING 1
+    if comb_bx is None and label_bx is None:
+        comb_bx = pd.DataFrame()
+        label_bx = [] 
+        #First define the colors for the observations.
+        obs_dict = dict(color='gray', linestyle='-',marker='x', linewidth=1.2, markersize=6.)
+        if plot_dict is not None:
+            #Whatever is not defined in the yaml file is filled in with the obs_dict here.
+            plot_kwargs = {**obs_dict, **plot_dict}
+        else:
+            plot_kwargs = obs_dict
+    else:
+        plot_kwargs = plot_dict
+    #For all, a column to the dataframe and append the label info to the list.
+    plot_kwargs['column'] = column
+    plot_kwargs['label'] = label
+    if df_reg is not None:
+        comb_bx[label] = df_reg[column+'_reg']
+    else:
+        comb_bx[label] = df[column] 
+        region_bx['set_regions']=df[region_name[0]]   #BEIMING 2 
+    label_bx.append(plot_kwargs)
+    
+    return comb_bx, label_bx,region_bx             #BEIMING 3
+
 def make_boxplot(comb_bx, label_bx, ylabel = None, vmin = None, vmax = None, outname='plot',
                  domain_type=None, domain_name=None,
                  plot_dict=None, fig_dict=None,text_dict=None,debug=False):
+
+    """Creates box-plot. 
+
+    Parameters
+    ----------
+    comb_bx: dataframe
```

Comment:
1. Needs to be a space on either side of the `:`
2. RHS should be the `type`, e.g. what you get if you used the `type()` function on the variable

```suggestion
    comb_bx : pandas.DataFrame
```

Reply:
thanks! I fixed that issue